### ISSUE TITLE
Needs New Update

### ISSUE THREAD

**Issue by PyrexianYT** [Umbranoxio/BeatSaberModInstaller#103]
There's a new update for Beat Saber and because of this, the old mods no longer work so we desperately need a new update for the mod installer soon.

**Comment by phiresky** [comment]
duplicate of #101. you should learn how github works before spamming people with emails.

**Comment by Baysen** [comment]
> duplicate of #101. you should learn how github works before spamming people with emails.

They want to report that their game doesn't work, not become a programer and send pull requests. Don't be so hard on them.

**Comment by fr34q** [comment]
I agree with @phiresky. They could spend a tiny amount of time to understand how the platform works. One of this kind of issue is fine, but there are four exactly identical issues now that don't contribute anything.

**Comment by notyoyoma** [comment]
@RemizZ Every internet forum ever has the concept of "Search for an existing thread before submitting a new one."

Of course, there is no need for the end users to learn source control just to submit an issue, but it is common sense to search beforehand.

**Comment by asolano86** [comment]
Not everyone knows how forums work - you guys need to drop your attitude towards people just looking for help.

**Comment by phiresky** [comment]
@asolano86  It's possible to have a tiny bit of respect when joining new communities, which implies spending a bit of time (like 5 minutes would have been enough) to try and understand it. 80 people get an email whenever someone creates an issue here.

One of the main reasons why open source projects die is because of the stress for maintainers that users cause who feel entitled to get more free stuff while doing absolutely nothing to contribute (not even a few minutes of their time). See [here](https://news.ycombinator.com/item?id=12000746) and [here](https://www.techrepublic.com/article/why-open-source-developers-are-burning-out-no-respect/).

**Comment by asolano86** [comment]
Politics like this make people afraid to ask for help. Just saying. Nobody is feeling entitled, not sure where you are getting that, all I see are people asking for help. Nothing entitled about that.

**Comment by phiresky** [comment]
None of these people bothered to help others by saying how they solved their problem (which at least some of them surely did by using [modsaber](https://github.com/lolPants/modsaber-installer) instead). All they did was waste the time of other people and immediately stop caring once they figured it out.

And yeah these aren't that bad but you really think "rly need it help" and "**we desperately need a new update**" don't sound like entitlement? The only one that sounds respectful is #101.

**Comment by notyoyoma** [comment]
In his first comment @phiresky linked to issue [#101](https://github.com/Umbranoxio/BeatSaberModInstaller/issues/101) In which @fr34q gives a [temporary workaround](https://github.com/Umbranoxio/BeatSaberModInstaller/issues/101#issuecomment-473580610).

He did help. He probably could have been more explicit and said "There's a solution here..." but even that wouldn't be correct, because there isn't a perfect fix yet.

Please also note: so far 4 issues have been raised in the last 5 days. That's frustrating for the devs, and (per the links @phiresky shared) runs the risk of hurting their motivation. We're all humans here, and nobody expects (devs or users) to be perfect. But courtesy is how we deal politely with each other, and we should expect that.

**Comment by asolano86** [comment]
God, all I said was to chill out a bit ~ looks like someones feelings were hurt. Calm down killer.

**Comment by notyoyoma** [comment]
@asolano86 You said:

> you guys need to drop your attitude...
> Politics like this...
> Is there no update on this mod yet?

You created an account on March 16, 2019, have contributed nothing, and are now:

1. Criticizing the people who spent their time to give you free stuff
2. Asking "Is there no update yet?"
3. Telling those people to "Calm down" and "Chill out"

Please, please, please, just learn to be courteous. The Open Source community is full of amazing people who will gladly take time to help new people. But you need to be willing to spend your own time to read and understand before making demands.

**Comment by asolano86** [comment]
Good god drop it, I already erased my comments, now drop it.

**Comment by Baysen** [comment]
@phiresky I don't think it was meant to be entitled. I am a developer myself, I know how frustrating cutomsers are.
Should they have looked for 10 seconds to see there is another issue like this? Yes. But this site can be confusing for newcomers.
Thank you for your work, which brings us all amazing joy, but please don't say they are entitled when they just wanted to report a broken game :)

**Comment by megalon** [comment]
Hey, I'm an admin on the Beat Saber Modding Group Discord server and an admin on ModSaber as well. Maybe I can clear a few things about this project.

The Mod Manager is just an application that gets the latest mods from www.modsaber.com

This project doesn't control any of the individual mods, it just gives you an easy way to download them.


The creator of this project (Umbranox, aka Umbranoxio as it shows on GitHub) was hired by Beat Games several months ago to bring official multiplayer into Beat Saber, and has since become a major part of the team. He even moved to Prague within the last few months (from Australia!) to work full time at the Beat Games studio.


Umbranox is also the owner and sole developer of www.scoresaber.com, the unofficial leaderboard website that tracks your scores for custom songs if you have the ScoreSaber mod installed (which he also developed). ScoreSaber is also responsible for the ranking system (pp), and has their own Discord server with a ranking team that decides which maps get ranked based on community suggestions and a ranking criteria.


That is a lot for one person to handle, and the Mod Manager project has clearly become a low priority. We've tried reaching out to Umbra to fix some of the issues or have him transfer the project to someone else, but as far as we can tell he's just too busy. As a result, the creator of ModSaber, lolPants, released the ModSaber Installer. It's a complete rewrite of the Mod Manager that accomplishes all of the same goals, fixes many issues, has more features, and is in active development.


https://github.com/lolPants/modsaber-installer/


You may as well unfollow the Mod Manager, because I don't see it getting any updates in the future.


If you want to follow active development of Beat Saber mods, get help, or just connect with the community, join the Discord server https://discord.gg/beatsabermods
You'll notice this is the same link from the "Help" tab within the Mod Manager.
Virtually all of the mod developers are in there, including Umbranox.


And if you want to join the ScoreSaber Discord
https://discord.gg/WpuDMwU


Apologies for any mistakes in this post, I'm currently on vacation and typing this up on my phone.

**Comment by phiresky** [comment]
Thank you for all the information @megalon

**Comment by Baysen** [comment]
Thank you @megalon ! Didn't even know there was a rewrite. This is also a big reason why open source projects die. The creator just abandons a project without even saying a word to the community he/she created, which is sad :(

**Comment by notyoyoma** [comment]
Thanks for the update and background @megalon.

**Comment by flamingtigers** [comment]
Thanks @megalon - much nicer and informative way of letting us know, opposed to a few other people that just want to jump down people's throat. I appreciate it.

**Comment by bearboman** [comment]
Thank you for the update and being a great community member/creator @megalon You are the kind of person places like this need and continue to thrive because of. Not some elitist jerk who comes by and says "Your so stupid for asking questions, stop being stupid cause we're better than you..." We appreciate that! Much obliged friend!

**Comment by phiresky** [comment]
@bearboman  Me and the others trying to maintain order here might have been jerks, but we were at least open about it and I still directed users to the solution of their problem (7 days ago). Every single person here with a new account and no contributions did not even realize they were being jerks by feeling entitled to complain while disregarding standard practice of the developer community, wasting everyone's time while not helping anyone. 

Being passive aggressive about it is also unhelpful.

**Comment by flamingtigers** [comment]
Not all of us are programmers, all we asked was for an update. Get over yourself @phiresky - you have been bitter since day one.

**Comment by bearboman** [comment]
@phiresky Trolls don't get to pretend they are helpful... even when they claim they're just "telling it like it is..." Maybe you should reexamine how you interact with the community as a whole and stop pretending that just because YOU have contributed code, you get to treat others like crap. That's the exact definition of elitist... weather you like it or not...

**Comment by Baysen** [comment]
@phiresky like I said, there was absolutely no entitlement in ANY of their posts. The word 'desperately' is not automatically entitlement. You seem to project some personal problems onto this. They all just want their game to work again and have fun. They are thankful for your work, otherwise they wouldn't even bother to make an account on a page they have nothing to do with.

**Comment by phiresky** [comment]
¯\\\_(ツ)_/¯ frustration about these things builds up over time, but in most cases it is more constructive to ignore it and to be respectful in order to not discourage people from contributing. If you've ever moderated a forum or maintained some public project i'm sure you know what i'm talking about. 

i chose this place to vent because it was clear from the start that none of the people that created an account here just to demand free stuff were planning on helping out anyways, so there was no harm to be done.

And just as a note, communities usually don't actually become better when you add more and more pure consumers. Just like adding leeches doesn't improve one's health and adding leechers doesn't improve a torrent. I'm not actually a contributor on this project either, but if I was I couldn't care less about people being offended by my directness unless they were actually trying to become a part of it.

@RemizZ It may seem to you that way but I ensure you that any open source maintainer you show the three duplicate issues here would interpret them just like I did, though they might be nicer about it. Imagine someone putting a box with cookies somewhere free for taking and then multiple people calling that person and saying "i desperately need you to make more cookies asap but this time please add vanilla" (not even bothering to spend 30sec to check that 4 other people have already made the same request and they could have +1'd that).

**Comment by notyoyoma** [comment]
@bearboman, @RemizZ there are now 5 issues. All people who didn't take a tiny amount of time to follow typical internet forum courtesy. That is a problem, and @phiresky was trying to address that problem. He was a bit harsh, but his sentiment was correct.

Before defending people for "just wanting to play their game", note that the original issue #101 has 23 👍. #102, #103, #104, and #105 have 9 👍 combined. So the vast majority of people were able to figure out how to use the system. The 4 people who submitted after #101 didn't take the time to read before submitting their issue. If we can give those 4 people a quick kick in the pants so next time they read for 5 minutes before adding their own issues, we should.

Edit: If you want to see what my idea of "a kick in the pants" looks like. [Here](https://github.com/Umbranoxio/BeatSaberModInstaller/issues/105#issuecomment-475912800). I'm a softie, but I probably haven't been through as much shit as @phiresky.

**Comment by MartijnVooges** [comment]
Geez what a waste of bytes this convo is. The guy should've done some basic research, yes. But how thirteen replies about it any better? I suggest we keep it at this, and show how professional GITters behave, by example :)

Thanks for the explanation, but the other repo's don't work for me, all 404. Others get that too?

**Comment by shootingwater** [comment]
After the new beat saber update the other day, even that alternative that worked no longer works. There is no working mod right now, but we get in trouble if we bring it up, which is why I am sure a lot of people haven't said anything.

**Comment by phiresky** [comment]
@MartijnVooges Yep, 404 for me as well. Googled it, see here for more info: https://www.reddit.com/r/beatsaber/comments/b904hj/modsaber_down/

**Comment by shootingwater** [comment]
> @shootingwater: `*ROLLEYES*`

What is your problem?

**Comment by phiresky** [comment]
another post here: https://www.reddit.com/r/beatsaber/comments/b91dme/modsaber_is_down_details_to_come/ not really more info though. looks like the maintainer of modsaber just rage quit

**Comment by shootingwater** [comment]
@phiresky: *ROLLEYES*

**Comment by megalon** [comment]
ModSaber has shut down, and the ModSaber Installer no longer works either.

Please join the Beat Saber Modding Group Discord server for the most up to date news in the #server-announcements channel
https://discord.gg/beatsabermods

At the time of this writing, this is the best way to install mods right now.

**LINKS**
Here's a link to all the mods that were on ModSaber as of this morning https://pastebin.com/Bcm85dpd
Here are manual instructions to installing mods: https://wiki.assistant.moe/beginners-guide

**Comment by phiresky** [comment]
for reference: some forks of the code base are still alive. This is the newest I could find:

https://github.com/blueYeti213/modsaber-installer

Sadly I can't find any fork of the modsaber repository and it does not exist in the wayback machine.

**Comment by shootingwater** [comment]
@phiresky - stop bitching already and acting so entitled. Fuck.

**Comment by phiresky** [comment]
According to google cache, the lolpants repo was on `Latest commit 6eee33a  6 days ago` so the above fork is not current.

Wayback machine has some of the releases and code. Not sure if it's enough to reconstruct the current version if it was necessary.

There should be three forks of the modsaber repo, but I can't find them. They might have gotten deleted with the main repo. I hope someone with a copy of the code steps up or lolpants changes their mind.

I've saved copies of most of what I found.

**Comment by phiresky** [comment]
@shootingwater 

just either shhhhh or do something useful please

**Comment by shootingwater** [comment]
@phiresky - either stop being a rude grumpy old man, or start being nicer to people that are really asking for help. I did contribute, just because I don't know coding or programming. I merely just came in today to say that the newest version, the alternative version that was working, isn't working anymore. I noticed nobody else had said it. That is contributing. Get off your high horse.

**Comment by phiresky** [comment]
Found the three forks of ModSaber itself:

https://github.com/megalon/ModSaber
https://github.com/artemiswkearney/ModSaber
https://github.com/realityquintupled/ModSaber

they are old though

**Comment by ghost** [comment]
I am deleting my account. There is ABSOLUTELY no reason why you have to be such a huge asshole and bully @phiresky - Github should ban you, and your account was reported, with the screenshots before you deleted everything you were telling me. People like you need to go jump off a bridge. Bye, no MOD is worth the type of harassment you get for asking for help.

**Comment by phiresky** [comment]
> I am deleting my account.

👍 thx

**Comment by shootingwaters** [comment]
@phiresky You are such a cunt.

**Comment by phiresky** [comment]
Just for the record for everyone else - I literally just told @shootingwater (now @shootingwaters lol) to please either contribute or be quiet instead of complaining meaninglessly. It's kinda hilarious

**Comment by shootingwaters** [comment]
@phiresky Go back to sucking your moms tits.

**Comment by phiresky** [comment]
Just for the record for everyone else - I literally just told @shootingwater (now @shootingwaters lol) to please either contribute or be quiet instead of complaining meaninglessly, and I stand by everything I said.. It's kinda hilarious.

@shootingwaters I've reported you for derogatory and sexist language, and threatening me with my personal address.
![image](https://user-images.githubusercontent.com/2303841/55518777-8f7ce480-5675-11e9-8504-37e178bb5db0.png)

![image](https://user-images.githubusercontent.com/2303841/55519335-a91f2b80-5677-11e9-85ea-c5712d0b0e98.png)

![image](https://user-images.githubusercontent.com/2303841/55519475-49755000-5678-11e9-87ca-df330ae129bc.png)

![image](https://user-images.githubusercontent.com/2303841/55519984-7d517500-567a-11e9-8220-3152a449bc62.png)

**Comment by ader86** [comment]
What a baby @phiresky - don't you have something better to be doing?

**Comment by phiresky** [comment]
nice alt account @ader86  @shootingwaters

**Comment by shootingwaters** [comment]
Right - what a whiny ass.

**Comment by shootingwaters** [comment]
I'll create as many as I want. You will never hear the end of me dumbass. You need to learn to stop being such a little bitch @phiresky

**Comment by shootingwaters** [comment]
PS: Your website phiresky.com/ is down.

**Comment by shootingwaters** [comment]
@phiresky Nobody cares. If you were such a great programmer - why is your app development website giving a 403 error? You are all over my ass for trying to state a mod didn't work. Get over yourself and grow up.

**Comment by phiresky** [comment]
and now threatening me?
![jk this image is totally fake](https://user-images.githubusercontent.com/2303841/55519522-7c1f4880-5678-11e9-989c-1496d840357f.png)

**Comment by shootingwaters** [comment]
Oh yes, great job at creating that one. Good thing I record everything I do on my computer. Dumbass. Try again.

**Comment by shootingwaters** [comment]
Creative though, good job.

**Comment by phiresky** [comment]
.

**Comment by shootingwaters** [comment]
Try me. Literally - I have so much security, cameras, I record everything I do on my computer through servers. So threaten me again, I dare you.

**Comment by shootingwaters** [comment]
You can keep deleting your threats all you want, I have it all recorded @phiresky

**Comment by phiresky** [comment]
.

**Comment by shootingwaters** [comment]
Professional of you, anymore where that came from? Once again, you are all talk, but not man enough to keep it up without deleting all your petty remarks.

**Comment by phiresky** [comment]
> implying i'm a man

**Comment by shootingwaters** [comment]
I have looked you up. I am not stupid.

**Comment by phiresky** [comment]
.

**Comment by shootingwaters** [comment]
South Africa ring a bell?

**Comment by shootingwaters** [comment]
Cobus Swart. 22 West Street East Town, Johannesburg, Gauteng. +27.0720878679 @phiresky - much more than a profile pic. Now get off my ass and I will delete all of my comments.

**Comment by phiresky** [comment]
.

**Comment by shootingwaters** [comment]
Proving to you that I know what I am talking about.

**Comment by shootingwaters** [comment]
@phiresky I am leaving now. No point in wasting my time on you - I will delete your personal information. I just wanted to prove a point. Take care.

**Comment by shootingwaters** [comment]
...and I just have to point out since you are lying in your post that I am threatening, remember, I have all this recorded. You can lie all you want, but we both know you are the only one making threats - if Github needs that recording from me I will gladly share it to prove what a liar you really are. Don't make this worse than it has to be, we both know you are lying, I have the literal proof.
